SQL assignment.txt

```diff
@@ -0,0 +1,1533 @@
+| 3002 |  1900.1 | 10/03/90 | 2007 | 1004 |
+| 3001 |   18.69 | 10/03/90 | 2008 | 1007 |
+| 3006 | 1098.16 | 10/03/90 | 2008 | 1007 |
++------+---------+----------+------+------+
+
+
+
+5) Find which salespeople currently have orders in the order table. 
+
+mysql> select snum, sname, city, comm  FROM salespeople WHERE snum IN ( SELECT DISTINCT snum  FROM orders);
+
++------+---------+-----------+------+
+| snum | sname   | city      | comm |
++------+---------+-----------+------+
+| 1001 | Peel    | London    |   12 |
+| 1002 | serres  | sanjose   |   13 |
+| 1003 | axeirod | new york  |   10 |
+| 1004 | motika  | london    |   11 |
+| 1007 | rifkin  | barcelona |   15 |
++------+---------+-----------+------+
+
+
+6) List names of all customers matched with the salespeople serving them.
+
+mysql> SELECT c.cname, s.sname  FROM customers AS c JOIN salespeople AS s ON c.snum = s.snum;
+
++----------+---------+
+| cname    | sname   |
++----------+---------+
+| hoffman  | Peel    |
+| giovamni | axeirod |
+| liu      | serres  |
+| grass    | serres  |
+| cisneras | Peel    |
+| cisneras | rifkin  |
+| pereira  | motika  |
++----------+---------+
+
+
+
+7) Find the names and numbers of all salespeople who have more than one customer.
+
+mysql> SELECT sname, snum FROM salespeople WHERE snum IN ( SELECT snum FROM customers GROUP BY snum HAVING COUNT(snum) > 1);
+
++--------+------+
+| sname  | snum |
++--------+------+
+| Peel   | 1001 |
+| serres | 1002 |
++--------+------+
+
+
+
+8) Count the orders of each of the salespeople and output the results in descending order.
+
+mysql> SELECT salespeople.snum, salespeople.sname, COALESCE(COUNT(orders.onum), 0) as order_count  FROM salespeople  LEFT JOIN orders ON salespeople.snum = orders.snum GROUP BY salespeople.snum, salespeople.sname ORDER BY order_count DESC;
+
++------+---------+-------------+
+| snum | sname   | order_count |
++------+---------+-------------+
+| 1001 | Peel    |           3 |
+| 1002 | serres  |           3 |
+| 1007 | rifkin  |           2 |
+| 1003 | axeirod |           1 |
+| 1004 | motika  |           1 |
+| 1008 | fran    |           0 |
++------+---------+-------------+
+
+
+
+9) List the customer table if and only if one or more of the customers in the Customer table are located in SanJose.
+
+mysql> select * from customers WHERE EXISTS (SELECT city FROM customers where city="sanjose");
+
++------+----------+---------+--------+------+
+| cnum | cname    | city    | rating | snum |
++------+----------+---------+--------+------+
+| 2001 | hoffman  | london  |    100 | 1001 |
+| 2002 | giovamni | rome    |    200 | 1003 |
+| 2003 | liu      | sanjose |    200 | 1002 |
+| 2004 | grass    | berlin  |    300 | 1002 |
+| 2006 | cisneras | london  |    100 | 1001 |
+| 2006 | cisneras | sanjose |    300 | 1007 |
+| 2007 | pereira  | rome    |    100 | 1004 |
++------+----------+---------+--------+------+
+
+
+
+10) Match salespeople to customers according to what city they live in.
```

Comment:
Match salespeople to customers , Customer Column should be there.

Reply:
resolved